src/Core/Cache/DefaultCacheFactory.php

```diff
@@ -114,20 +117,52 @@ protected function isPHPFilesSupported()
     }
 
     /**
-     * @param string $class
-     * @param array $args
-     * @return CacheInterface
+     * Creates an object with a PSR-16 interface, usually from a PSR-6 class name
+     *
+     * Quick explanation of caching standards:
+     * - Symfony cache implements the PSR-6 standard
+     * - Symfony provides adapters wrap a PSR-6 backend with a PSR-16 interface
+     * - Silverstripe uses the PSR-16 to interface with caches
+     * - Psr\SimpleCache\CacheInterface is the php interface of the PSR-16 standard
+     *
+     * Further reading:
+     * - https://symfony.com/doc/current/components/cache/psr6_psr16_adapters.html#using-a-psr-6-cache-object-as-a-psr-16-cache
+     * - https://github.com/php-fig/simple-cache
      */
-    protected function createCache($class, $args)
+    protected function createCache(string $class, array $args, bool $useInjector = true): CacheInterface
     {
-        /** @var CacheInterface $cache */
-        $cache = Injector::inst()->createWithArgs($class, $args);
-
-        // Assign cache logger
-        if ($this->logger && $cache instanceof LoggerAwareInterface) {
-            $cache->setLogger($this->logger);
+        $classIsPsr6 = is_a($class, CacheItemPoolInterface::class, true);
+        $classIsPsr16 = is_a($class, CacheInterface::class, true);
+        if (!$classIsPsr6 && !$classIsPsr16) {
+            throw new InvalidArgumentException("class $class must be PSR-6 or PSR-16 compatible");
         }
-
-        return $cache;
+        if ($classIsPsr6) {
+            if ($useInjector) {
+                // Injector is used for in most instances to allow modification of the cache implementations
+                $psr6Cache = Injector::inst()->createWithArgs($class, $args);
+            } else {
+                // ManifestCacheFactory cannot use Injector because config is not available at that point
```

Comment:
Adding a whole branch of extra code just for `ManifestCacheFactory` doesn't feel right. It used to implement its own `createCache()` method - that still feels like the correct way to handle this situation.

Reply:
We'll end up with code duplication if we do that, e.g. adding the logger.  I'd rather have all the logic in one place.

I've tidied this up by creating a new private method `instantiateCache()` that handles the creation method. Doing this removed a fair bit of code

Reply:
The logger was never being added from `ManifestCacheFactory::createCache()` anyway, so I'm not worried about duplicating that. So ultimately the duplication you'd have is:
1. Check if the class uses one of the PSR6 or PSR16 interfaces
    - Throw an exception if not
1. Instantiate the class, which we'd be doing without injector anyway so that's not duplicated
1. Wrapping the class, which we'd be doing without injector anyway so that's not duplicated

I think it actually ends up a lot cleaner letting `ManifestCacheFactory` handle its own `createCache()` logic.

Reply:
> The logger was never being added from ManifestCacheFactory::createCache() anyway, so I'm not worried about duplicating that

You but now it does, what is an improvement

For instance we can now do something like set an `SS_MANIFESTCACHE` env variable = `MyCustomLogger`

where 
`class MyCustomerLogger implements CacheInterface, LoggerAwareInterface`

And it'll automatically add whatever the registered logger is during instantiation.

It's not just the logger code that we'd need duplicate if separate this out to ManifestCacheFactory, it's also the Psr16Cache wrapper code that would need to be duplicated

I think it's good as it current is

Reply:
I think looking at the diff was making this look a lot more convoluted than it was - looking at just the final code outside of the context of a diff, it looks good as is.